Gems/Terrain/Assets/Shaders/Terrain/TerrainDetailClipmapGenerationPass.azsl

```diff
@@ -23,102 +28,124 @@ ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
     RWTexture2DArray<float> m_detailOcclusionClipmaps;
 }
 
-[numthreads(32,32,1)]
+[numthreads(THREAD_NUM_X,THREAD_NUM_Y,1)]
 void MainCS(
-    uint3 dispatchThreadID : SV_DispatchThreadID)
+    uint3 groupID : SV_GroupID,
+    uint3 groupThreadID : SV_GroupThreadID)
 {
-    uint2 pixelPosition = dispatchThreadID.xy;
-
     for (uint clipmapLevel = 0; clipmapLevel < TerrainSrg::m_clipmapData.m_detailClipmapStackSize; ++clipmapLevel)
     {
-        uint3 texelIndex = uint3(dispatchThreadID.xy, clipmapLevel);
+        for (uint updateRegionIndex = 0; updateRegionIndex < 6; ++updateRegionIndex)
```

Comment:
Suggestion: Add a comment and named constant to help explain why there's 6 regions.